src/DynamicData/List/Internal/TransformMany.cs

```diff
@@ -101,7 +101,7 @@ public IObservable<IChangeSet<TDestination>> Run()
                 return _source.Transform(item => new ManyContainer(_manyselector(item).ToArray()), true)
                     .Select(changes =>
                     {
-                        var destinationChanges = new ChangeSet<TDestination>(new DestinationEnumerator(changes, _equalityComparer));
+                        var destinationChanges = new DestinationEnumerator(changes, _equalityComparer);

```

Comment:
Passed directly as an enumerable to Clone().  No need to make a changeset to hold the results first.